Common/MemoryUtil.cpp

```diff
@@ -157,11 +158,21 @@ void *AllocateExecutableMemory(size_t size) {
 		ptr = VirtualAlloc(0, size, MEM_RESERVE | MEM_COMMIT, prot);
 #endif
 	}
-#else
-	static char *map_hint = 0;
+	if (!ptr) {
+		ERROR_LOG(MEMMAP, "Failed to allocate executable memory (%d)", (int)size);
+	}
+	return ptr;
+}
+
+#else  // Non-Windows platforms
+
+void *AllocateExecutableMemory(size_t size) {
+	static char *map_hint = nullptr;
+
 #if PPSSPP_ARCH(AMD64)
 	// Try to request one that is close to our memory location if we're in high memory.
 	// We use a dummy global variable to give us a good location to start from.
+	// TODO: Should we also do this for ARM64?
```

Comment:
I think we mainly do this for RIP-relative addressing reasons, right?  It may not be as necessary for arm64.

-[Unknown]